src/main/kotlin/org/dashj/platform/dpp/util/Cbor.kt

```diff
@@ -131,139 +139,35 @@ object Cbor {
 
         sortedKeys.forEach { key ->
             val value = obj[key]
-            if (value is Map<*, *>) {
-                if (innerMapBuilder != null && innerMapBuilder is MapBuilder<*>) {
-                    writeJSONObject(obj[key] as Map<String, Any?>, mapBuilder, baos, innerMapBuilder.putMap(key))
-                } else {
-                    writeJSONObject(obj[key] as Map<String, Any?>, mapBuilder, baos, mapBuilder.putMap(key))
-                }
-            } else {
-                val builder: AbstractBuilder<*> = innerMapBuilder ?: mapBuilder
-                if (value is List<*>) {
-                    if (builder is MapBuilder<*>) {
-                        addJSONArray(value, mapBuilder, baos, builder.putArray(key))
-                    } else if (builder is ArrayBuilder<*>) {
-                        addJSONArray(value, mapBuilder, baos, builder.addArray())
-                    }
-                } else if (builder is MapBuilder<*>) {
-                    addValueToMapBuilder(builder, key, value)
-                } else if (builder is ArrayBuilder<*>) {
-                    addValueToArrayBuilder(value, builder)
-                }
+            when (value) {
+                is Map<*, *> -> writeJSONObject(obj[key] as Map<String, Any?>, builder.putMap(key))
+                is List<*> -> writeJSONArray(value, builder.putArray(key))
+                else -> addValueToMapBuilder(builder, key, value)
```

Comment:
The original code from years ago was overly complicated.  It was passing parameters that were not used and making checks for classes with `is` that could never be `true`